tests/unit/parsers/test.manifestjson.js

```diff
@@ -3179,6 +3179,47 @@ describe('ManifestJSONParser', () => {
       );
     });
 
+    // See: https://bugzilla.mozilla.org/show_bug.cgi?id=1860304
+    it.each([
+      { scripts: ['bg.js'] },
+      { page: 'bg.html' },
+      { scripts: [], page: 'bg.html' },
+    ])(
+      'emits a warning when background.service_worker is being used in combination with %o',
+      (backgroundProps) => {
+        const linter = new Linter({ _: ['bar'] });
+        const json = validManifestJSON({
+          background: {
+            service_worker: 'background_worker.js',
+            ...backgroundProps,
+          },
+        });
+        // eslint-disable-next-line no-unused-vars
+        const manifestJSONParser = new ManifestJSONParser(
+          json,
+          linter.collector,
+          {
+            io: {
+              files: { 'background_worker.js': '', 'bg.js': '', 'bg.html': '' },
+            },
+          }
+        );
+
+        expect(linter.collector.errors).toEqual([]);
+        // expect(manifestJSONParser.isValid).toEqual(true);
```

Comment:
@willdurand I believe you may have commented this while you were figuring out where the strict behavior related to the additional properties was coming from and forgot to uncomment it, it does currently pass as expected if uncommented.

would you mind to also move it after the assertion on the warnings along with uncommenting it? (in the past it was often annoying to get a failure from this assertion without having hit the more detailed assertion that follows it in this test and so I usually put this assertion at the end to make it easier to investigate regressions hit in ci jobs).